addons/binding/org.openhab.binding.helios/src/main/java/org/openhab/binding/helios/handler/HeliosHandler27.java

```diff
@@ -0,0 +1,450 @@
+    public static final String IP_ADDRESS = "ipAddress";
+    public static final String OPENHAB_IP_ADDRESS = "openHABipAddress";
+    public static final String OPENHAB_PORT_NUMBER = "openHABportNumber";
+
+    private final Logger logger = LoggerFactory.getLogger(HeliosHandler27.class);
+
+    private static final String SUBSCRIPTION_PERIOD = "PT1H";
+
+    private JAXBContext context = null;
+    private Endpoint endpoint = null;
+    private NotificationProducer notificationProducer = null;
+    private PausableSubscriptionManager subscription = null;
+    private GregorianCalendar currentTime;
+    private GregorianCalendar terminationTime;
+    private W3CEndpointReference subscriptionReference;
+    private String subscriptionID;
+    private SOAPEvent previousEvent = null;
+
+    public static final String HELIOS_URI = "http://www.2n.cz/2013/event";
+    public static final String DIALECT_URI = "http://www.2n.cz/2013/TopicExpression/Multiple";
+    public static final String WSN_URI = "http://docs.oasis-open.org/wsn/b-2";
+
+    public static final QName TOPIC_EXPRESSION = new QName(WSN_URI, "TopicExpression");
+    public static final QName INITIAL_TERMINATION_TIME = new QName(WSN_URI, "InitialTerminationTime");
+    public static final QName MAXIMUM_NUMBER = new QName(HELIOS_URI, "MaximumNumber");
+    public static final QName SIMPLE_MESSAGES = new QName(HELIOS_URI, "SimpleMessages");
+    public static final QName START_TIME_STAMP = new QName(HELIOS_URI, "StartTimestamp");
+    public static final QName START_RECORD_ID = new QName(HELIOS_URI, "StartRecordId");
+
+    public HeliosHandler27(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            // 2N.cz has not yet released the automation part of the Helios IP HTTP/SOAP based API. Only the
+            // notification part has been documented, so for now there is nothing to do
+            // here
+            logger.debug("The Helios IP is a read-only device and can not handle commands");
+        }
+    }
+
+    public String getSubscriptionID() {
+        return subscriptionID;
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Helios IP handler.");
+        List<Handler> handlerChain = new ArrayList<Handler>();
+        handlerChain.add(new SOAPActionHandler());
+
+        try {
+            context = JAXBContext.newInstance(SOAPObjectFactory.class, SOAPEvent.class, SOAPKeyPressed.class,
+                    SOAPCallStateChanged.class, SOAPCodeEntered.class, SOAPCardEntered.class, SOAPDeviceState.class);
+        } catch (JAXBException e) {
+            logger.error("An exception occurred while setting up the JAXB Context factory: {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
```

Comment:
This isn't a valid combination. As something is tremendously wrong here, you might consider to throw some RuntimeException from this method.